compliance_suite/cli.py

```diff
@@ -38,41 +38,55 @@ def report(server: str,
 
     Args:
         server (str): The server URL on which the compliance suite will be run. Format - https://<url>/
-        version (str): The compliance suite will be run against this TES version. Example - "1.0.0"
+        version (List[str]): The compliance suite will be run against this TES version. Example - "1.0.0"
         tag (List[str]): The list of the tags for which the compliance suite will be run. Default - All
         output_path (str): The output path to store the JSON compliance report
         serve (bool): If true, runs a local server and displays the JSON report in webview
-        port (int): Set the local server port. Default - 16800
+        port (int): Set the local server port. Default - 15800
         uptime (int): The local server duration in seconds. Default - 3600 seconds
     """
 
     if server is None:
         raise Exception("No server provided. Please provide a server URL.")
 
-    if version is None:
+    if not versions:
         raise Exception("No version provided. Please provide a version.")
 
     tag = [val.lower() for val in tag]      # Convert the tags into lowercase to allow case-insensitive tags
     logger.info(f"Input tag is - {tag}")
-    job_runner = JobRunner(server, version, tag)
-    job_runner.run_jobs()
 
-    json_report = job_runner.generate_report()
+    versions = list(versions)
+    versions.sort()
+    logger.info(f"Input version(s) - {versions}")
 
-    # Store the report in given output path
-    if output_path is not None:
-        logger.info(f"Writing JSON Report on directory {output_path}")
-        with open(os.path.join(output_path, "report.json"), "w+") as output:
-            output.write(json_report)
+    for version in versions:
+        job_runner = JobRunner(server, version, tag)
+        job_runner.run_jobs()
+
+        json_report = job_runner.generate_report()
+
+        # Store the report in given output path
+        if output_path is not None:
+            logger.info(f"Writing JSON Report on directory {output_path}")
+            with open(os.path.join(output_path, f"report-tes-{version}.json"), "w+") as output:
+                output.write(json_report)
```

Comment:
If a test is executed by any of the versions, we should not repeat it again as the result would be the same every time.
We'll need to explore how to store the state of executed tests.

Another thing to note, when generating the report, we'll need the copy and append the common test result of the first version to the other versions. Otherwise, they'll miss out on tests.